### PR TITLE
Stabilize `nonnull_slice_from_raw_parts`

### LINKED ISSUE #71941: Tracking Issue for `NonNull<[T]>::slice_from_raw_parts`
This is a tracking issue for the `NonNull<[T]>::slice_from_raw_parts` method proposed in PR https://github.com/rust-lang/rust/pull/71940.
The feature gates for the issue are `#![feature(nonnull_slice_from_raw_parts)]`, and `#![feature(const_nonnull_slice_from_raw_parts)]` for using as a `const fn`.

### API

```rust
impl<T> NonNull<[T]> {
  pub const fn slice_from_raw_parts(data: NonNull<T>, len: usize) -> Self;
}
```

Docs: https://doc.rust-lang.org/stable/std/ptr/struct.NonNull.html#method.slice_from_raw_parts

### Steps

- [x] Implementation: https://github.com/rust-lang/rust/pull/71940
- [x] FCP: https://github.com/rust-lang/rust/issues/71941#issuecomment-1065939514
- [ ] Stabilization PR

### PR DESCRIPTION
**Pull request by JohnTitor**
FCP is done: https://github.com/rust-lang/rust/issues/71941#issuecomment-1100910416
Note that this doesn't const-stabilize `NonNull::slice_from_raw_parts` as `slice_from_raw_parts_mut` isn't const-stabilized yet. Given #67456 and #57349, it's not likely available soon, meanwhile, stabilizing only the feature makes some sense, I think.

Closes #71941

### CODE REVIEW

**Comment by JohnTitor** [comment]
r? rust-lang/libs-api

**Comment by JohnTitor** [comment]
Seems it doesn't work, r? @m-ou-se maybe?

File: library/core/src/ptr/non_null.rs [hunk 1]
```diff
@@ -480,7 +478,7 @@ impl<T> NonNull<[T]> {
     ///
     /// (Note that this example artificially demonstrates a use of this method,
     /// but `let slice = NonNull::from(&x[..]);` would be a better way to write code like this.)
-    #[unstable(feature = "nonnull_slice_from_raw_parts", issue = "71941")]
+    #[stable(feature = "nonnull_slice_from_raw_parts", since = "1.64.0")]
     #[rustc_const_unstable(feature = "const_nonnull_slice_from_raw_parts", issue = "71941")]
```
Comment (m-ou-se) [inline library/core/src/ptr/non_null.rs:482]: Maybe it's better if we close #71941 and move tracking of the const part to the const_slice_from_raw_parts issue.
```suggestion
    #[rustc_const_unstable(feature = "const_slice_from_raw_parts", issue = "67456")]
```

Comment (JohnTitor) [reply library/core/src/ptr/non_null.rs]: Sounds good! Updated as suggested (sorry for the delay, I was sick these days).

**Comment by est31** [comment]
:wave: Hello, I'm writing this comment in this **stabilization PR** to notify you, the authors of this PR, that https://github.com/rust-lang/rust/pull/100591 has been merged, which implemented a change in how features are stabilized.

Your PR has been filed before the change, so will likely require modifications in order to comply with the new rules. I recommend you to:

1. rebase the PR onto latest master, so that uses of the placeholder are possible.
2. replace the version numbers in the PR with the placeholder `CURRENT_RUSTC_VERSION`. For language changes, this means the version numbers in `accepted.rs` (example: 4caedbae869855a19ca49455e2cd338ae3b8c1ef). For library changes, this means the `since` fields (example e576a9b554f24a26c36c04b41c901636d190605b).

That's it! The `CURRENT_RUSTC_VERSION` placeholder will, as part of the release process, be replaced with the version number that the PR merged for. It can be used anywhere in `rust-lang/rust`, not just accepted.rs and the since fields.

If you have any questions, feel free to drop by the [zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/libs.20stabilization.20placeholder), or ping me directly in this PR's thread. Thanks! :wave:

**Comment by the8472** [comment]
FCP has happened in the tracking issue, review comment has been addressed, so I think this can be merged. I'll update #67456

@bors r=m-ou-se,the8472

**Comment by ChrisDenton** [comment]
I think bors was ignoring people that day. Let's try this again.

@bors r=m-ou-se,the8472

**Comment by ChrisDenton** [comment]
Let's try again...

@bors r=m-ou-se,the8472

**Pull request merged**